packages/buidler-core/src/internal/buidler-evm/provider/node.ts

```diff
@@ -1437,6 +1470,27 @@ If you are using a wallet or dapp, try resetting your wallet's accounts.`
     return parentTd.add(difficulty);
   }
 
+  private async _runOnBlockContext<T>(
+    blockNumber: BN | null,
```

Comment:
`null` here would mean `pending`, which for now is the latest block, so no modification is needed.